lib/ide/src/executor/global.rs

```diff
@@ -0,0 +1,56 @@
+//! This module defines a shared, global state that stores a task spawner.
+//!
+//! It is expected that as a part of one-time initialization routine, IDE shall
+//! call `set_spawner` to define a global spawner. Once that is done, the whole
+//! codebase can use `spawn` function to schedule new tasks on the global
+//! executor.
+//!
+//! The executor responsible for spawning tasks is expected to remain alive
+//! indefinitely, at least until the last `spawn` call has been made.
+
+
+use crate::prelude::*;
+
+use futures::task::LocalSpawnExt;
+use futures::task::LocalSpawn;
+
+/// Global spawner handle.
+///
+/// It should be set up once, as part of the IDE initialization routine and
+/// remain accessible indefinitely.
+static mut SPAWNER: Option<Box<dyn LocalSpawn>> = None;
```

Comment:
Could we use here [lazy-static](http://rust-lang-nursery.github.io/lazy-static.rs/lazy_static/index.html) instead?

Reply:
Moreover, I think it would be beneficial to tell in docs here what is a spawner. I read almost the whole file and still I don ot understand what is a spawner and what is its relation to executor. I know it "spawns tasks", but what it really means? What is an example of such task, what is the difference between spawning a task and calling a function, etc.